.soft -> .on_failure

jez opened the pull request:
<!-- (optional) Explain your change, focusing on the details of the solution. This is a great place to call out user-visible changes. -->

### Motivation
<!-- Why make this change? Describe the problem, not the solution. This can also be a link to an issue. -->

This is the first of a handful of changes to fully support `.on_failure` in the
runtime.

### Commit summary

This change is largely a codemod, so I've included my codemod'ing commands and
broke it into commits.

- `git grep --line '\.soft(' | grep -v 'Opus::Error\.soft' | fileline | multi-sub --global '\.soft' '.on_failure'` (a0c33f2d6)

- `git grep --line -w 'soft' | grep -v 'Opus::Error\.' | grep -v 'soft.assert' | fileline | multi-sub --global 'soft' 'on_failure'` (e90fe281c)

- `manual: git grep -w --line soft | fileline | vim -` (adfdc0f3e)

- **Stop raising for on_failure** (fa2fa8ede)

  This is the change where we actually stop raising in the runtime. The API
  isn't quite final yet (specifically, we're going to relax `notify:` from
  being required), but I wanted to make it stop raising so that I could get the
  confidence that the tests pass after this codemod.

- **Fix test that broke while it was skipped** (4fb21e4b5)

  This test was something that should have been fixed in a previous change
  (either the one to factor out move our `signature.soft_notify` logic into
  pay-server, or the one to add `T::Configuration.soft_notify_handler`).
  Regardless, now that it's not skipped, I fixed the test.



### Test plan
<!-- If you did not write tests for this change, replace the message below explaining why not. Why we should be confident this change is correct? If you changed the website, please include a screenshot of the proposed changes. -->

See included automated tests.

jez commented:
⚠️ Please don't merge this PR for me ⚠️ 

It's based of a non-master branch rn.

manhhung741 commented:
This change could break our code when it is merged/ pushed to RubyGems. Could you keep me in the loop when it happens? 
I think we need a way to communicate big changes/breaking changes in general (the `rbi/`, `rbis/`  change was another example)

jez commented:
@manhhung741 how can this break your code? `.soft` currently raises in the runtime.

manhhung741 commented:
We will also need to refactor the `.soft` in our code to `.on_failure` do
we?

On Fri, Jun 28, 2019 at 11:27 PM Jake Zimmerman <notifications@github.com>
wrote:

> @manhhung741 <https://github.com/manhhung741> how can this break your
> code? .soft currently raises in the runtime.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/sorbet/sorbet/pull/1141?email_source=notifications&email_token=AAFH4AOVXM7GRG4EVBAYFR3P43QBNA5CNFSM4H4JQD3KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODY3RB6Y#issuecomment-506925307>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAFH4AJFBUDMNGQHE2G27WDP43QBNANCNFSM4H4JQD3A>
> .
>

jez commented:
> We will also need to refactor the `.soft` in our code to `.on_failure` do we?

We removed the ability for people to use `.soft` before we open sourced sorbet. If you're using `.soft`, you'll find that it's already broken for you on master, even without this PR. We removed it because we weren't comfortable with the API, and wanted to rethink it before people started using it. It's possible you're the only company from the private beta that's using `.soft`.

manhhung741 commented:
Oof. Ok. I guess I should remove them. It’s interesting that it doesn’t
raise any syntax error currently with the `.soft` existing. B

We also override runtime error to logging only so it’s not “breaking” the
run time.

On Fri, Jun 28, 2019 at 11:31 PM Jake Zimmerman <notifications@github.com>
wrote:

> We will also need to refactor the .soft in our code to .on_failure do we?
>
> We removed the ability for people to use .soft before we open sourced
> sorbet. If you're using .soft, you'll find that it's already broken for
> you on master, even without this PR. We removed it because we weren't
> comfortable with the API, and wanted to rethink it before people started
> using it. It's possible you're the only company from the private beta
> that's using .soft.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/sorbet/sorbet/pull/1141?email_source=notifications&email_token=AAFH4AI2SOILIQALN7MSX6TP43QTPA5CNFSM4H4JQD3KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODY3RDZA#issuecomment-506925540>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAFH4AKANQT54LFQ6KLVC4DP43QTPANCNFSM4H4JQD3A>
> .
>
